File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/filterdata.rb
module Asposetasksjava
  module FilterData
    def initialize()
        # Reading Filter Definition Data
        get_filter_definition_data()
    end

    def get_filter_definition_data()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        task_filters = project.getTaskFilters().toList()

        puts "Task Filters Count: " + task_filters.size().to_s
        puts "All Tasks: " + task_filters.get(0).getName().to_s
        puts "Task Item: " + task_filters.get(0).getFilterType().to_s
        puts "Task Filters Show In Menu: " + task_filters.get(0).getShowInMenu().to_s
        puts "Task filter ShowRelatedSummaryRows: "  + task_filters.get(0).getShowRelatedSummaryRows().to_s

        rsc_filters = project.getResourceFilters().toList()

        puts "Project.ResourceFilters count: " +  rsc_filters.size().to_s
        puts "Resource Filter Item Type: Item.ResourceType: "  + rsc_filters.get(0).getFilterType().to_s
        puts "Resource filter ShowInMenu"  + rsc_filters.get(0).getShowInMenu().to_s
        puts "Resource filter ShowRelatedSummaryRows: " + rsc_filters.get(0).getShowRelatedSummaryRows().to_s
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/FilterData.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;

class FilterData{

    public static function run($dataDir=null){
        # Reading Filter Definition Data
        FilterData::get_filter_definition_data($dataDir);
    }

    public static function get_filter_definition_data($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $task_filters = $project->getTaskFilters()->toList();

        print "Task Filters Count: " . (string)$task_filters->size().PHP_EOL;//.to_s
        print "All Tasks: " . (string)$task_filters->get(0)->getName().PHP_EOL;//.to_s
        print "Task Item: " . (string)$task_filters->get(0)->getFilterType().PHP_EOL;//.to_s;
        print "Task Filters Show In Menu: " . (string)$task_filters->get(0)->getShowInMenu().PHP_EOL;//.to_s
        print "Task filter ShowRelatedSummaryRows: "  . (string)$task_filters->get(0)->getShowRelatedSummaryRows().PHP_EOL;//.to_s

        $rsc_filters = $project->getResourceFilters()->toList();

        print "Project.ResourceFilters count: " .  (string)$rsc_filters->size().PHP_EOL;//.to_s
        print "Resource Filter Item Type: Item.ResourceType: "  . (string)$rsc_filters->get(0)->getFilterType().PHP_EOL;//.to_s
        print "Resource filter ShowInMenu"  . (string)$rsc_filters->get(0)->getShowInMenu().PHP_EOL;//.to_s
        print "Resource filter ShowRelatedSummaryRows: " . (string)$rsc_filters->get(0)->getShowRelatedSummaryRows().PHP_EOL;//.to_s
    }

}
?>